libraries/AP_HAL_ChibiOS/AnalogIn.cpp

```diff
@@ -64,6 +64,14 @@ const AnalogIn::pin_info AnalogIn::pin_config[] = HAL_ANALOG_PINS;
 
 #define ADC_GRP1_NUM_CHANNELS   ARRAY_SIZE(AnalogIn::pin_config)
 
+#if defined(STM32H7) || defined(STM32F3) || defined(STM32G4)
+// on H7 we use 16 bit ADC transfers, giving us more resolution. We
+// need to scale by 1/16 to match the 12 bit scale factors in hwdef.dat
+#define ADC_BOARD_SCALING (1.0/16)
```

Comment:
Should we shift instead?

Reply:
no, that would lose the extra 4 bits of precision